src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -675,6 +712,285 @@ GroupPrincipal getGroupPrincipal(Member group) {
+    // start point in the id path.
+    StartPoint startPoint = null;
+    int startPointIndex = -1;
+    for (StartPoint sp : this.startPoints) {
+      String startPointId = String.valueOf(sp.getNodeId());
+      if (startPointId.equals(objectId)) {
+        return escapeParam(sp.getName()) + ":" + startPointId;
+      } else {
+        startPointIndex = ids.indexOf(startPointId);
+        if (startPointIndex > -1) {
+          startPoint = sp;
+          break;
+        }
+      }
+    }
+    // If the result item is not contained under a configured
+    // start point, skip it.
+    if (startPointIndex == -1) {
+      log.log(Level.FINE, "No start point in path: " + ids);
+      return null;
+    }
+
+    // Get the list of names defining the path to the search result
+    // item from the LocationPathString element.
+    List<String> names = getXmlSearchNames(resultElement);
+    if (names.size() != ids.size()) {
+      log.log(Level.FINE, "Mismatch in names/ids in search results");
+      return null;
+    }
+
+    // The first element in the doc id is the start point; use
+    // the value from the config file.
+    StringBuilder docId = new StringBuilder();
+    docId.append(escapeParam(startPoint.getName()));
+    for (int i = startPointIndex + 1; i < names.size(); i++) {
+      docId.append("/").append(escapeParam(names.get(i)));
+    }
+    docId.append("/")
+        .append(paramEscaper.escape(getTextContent(resultElement, "OTName")))
+        .append(":")
+        .append(objectId);
+
+    return docId.toString();
+  }
+
+  /* LocationPathString contains a ':' separated list of object
+   * names, not including the name of the search result object.
+   * Example:
+   * <LocationPathString>Enterprise:Folder 1:Project 1</LocationPathString>
+   */
+  @VisibleForTesting
+  List<String> getXmlSearchNames(Element resultElement) {
+    return Splitter.on(":").trimResults().omitEmptyStrings()
+        .splitToList(getTextContent(resultElement, "LocationPathString"));
+  }
+
+  /* OTLocation contains a list of object ids, starting from the
+   * root, defining the path to (and including) the search result object.
+   * Example:
+   * <OTLocation><![CDATA[2000 1234 5678 -5678 9012]]></OTLocation>
+   *
+   * This method returns a modifiable list.
+   */
+  @VisibleForTesting
+  List<String> getXmlSearchIds(Element resultElement) {
+    List<String> ids = new ArrayList<>(
+        Splitter.on(" ").omitEmptyStrings().trimResults()
+        .splitToList(getTextContent(resultElement, "OTLocation")));
+    ListIterator<String> iterator = ids.listIterator();
+    while (iterator.hasNext()) {
+      String id = iterator.next();
+      Long idn = Longs.tryParse(id);
+      if (idn == null) {
+        log.log(Level.FINER, "Invalid id '{0}' in id list {1}",
+            new Object[] {id, ids});
+        return Collections.emptyList();
+      } else if (idn < 0) {
+        iterator.remove(); // Remove volume ids, e.g. Projects.
```

Comment:
Could the list be empty at this point if the search result is the project volume itself (sort of hard to arrange, perhaps, whereas getting the project should be easy but in that case we'll have the full path from its volume).

Let's just check, to avoid silently skipping things, which hard to support.